File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main(void) {
    int result = 0;
    stack<char> s;
    string parenthesis;
    cin >> parenthesis;

    for (int i = 0; i < parenthesis.length(); i++) {
        if (parenthesis[i] == '(') {
            if (parenthesis[i + 1] == ')') {
                result += s.size();
                i += 1;
            }
            else {
                s.push(parenthesis[i]);
            }
        } else {
            result++;
            s.pop();
        }
    }
    cout << result << '\n';
    return 0;
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
parenthesis = input()

answer = 0
stack = []
for i in range(len(parenthesis)) :
    if (parenthesis[i] == '(') :
        stack.append('(')
    else :
        if (parenthesis[i - 1] == '(') :
            stack.pop()
            answer += len(stack)
        else :
            stack.pop()
            answer += 1
            
print(answer)